src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionItem.cs

```diff
@@ -19,12 +20,12 @@ public sealed class PackageCollectionItem : PackageIdentity
         /// <summary>
         /// Installed package references.
         /// </summary>
-        public List<IPackageReferenceContextInfo> PackageReferences { get; }
+        public IReadOnlyCollection<IPackageReferenceContextInfo> PackageReferences { get; }
 
         public PackageCollectionItem(string id, NuGetVersion version, IEnumerable<IPackageReferenceContextInfo> installedReferences)
             : base(id, version)
         {
-            PackageReferences = installedReferences?.ToList() ?? new List<IPackageReferenceContextInfo>();
+            PackageReferences = installedReferences?.ToArray() ?? Array.Empty<IPackageReferenceContextInfo>();
```

Comment:
Why did you do this change?

Reply:
We don't need to allocate a new list if there are no references.

Reply:
I meant the ToList -> ToArray change, in general it is better to use ToList over ToArray

Reply:
If I understood correctly Array.Empty doesn't allocate a memory so it doesn't stress GC, is it right?

Reply:
`Array.Empty<T>()` does allocate memory, but it's idempotent, meaning that first call allocates but subsequent calls do not.